kong/plugins/rate-limiting/handler.lua

```diff
@@ -28,10 +28,14 @@ local function get_identifier(conf)
     end
   elseif conf.limit_by == "credential" then
     identifier = ngx.ctx.authenticated_credential and ngx.ctx.authenticated_credential.id
+  elseif conf.limit_by == "http_header" then
+    identifier = ngx.req.get_headers()[conf.http_header]
   end
 
   if not identifier then identifier = ngx.var.remote_addr end
 
+  ngx_log(ngx.DEBUG, "Using identifier: ", identifier)
```

Comment:
If we want to keep this log entry, we should prepend it with a unique identifier for context.